bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/handler/OpenWebNetBridgeHandler.java

```diff
@@ -0,0 +1,577 @@
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OpenWebNetBridgeHandler} is responsible for handling communication with gateways and handling events.
+ *
+ * @author Massimo Valla - Initial contribution
+ */
+@NonNullByDefault
+public class OpenWebNetBridgeHandler extends ConfigStatusBridgeHandler implements OpenListener {
+
+    private final Logger logger = LoggerFactory.getLogger(OpenWebNetBridgeHandler.class);
+
+    private static final int GATEWAY_ONLINE_TIMEOUT = 20; // (sec) Time to wait for the gateway to become connected
+    private static final int CONFIG_GATEWAY_DEFAULT_PORT = 20000;
+    private static final String CONFIG_GATEWAY_DEFAULT_PASSWD = "12345";
+    private static final boolean CONFIG_GATEWAY_DEFAULT_DISCVERY_ACTIVATION = false;
+
+    public final static Set<ThingTypeUID> SUPPORTED_THING_TYPES = OpenWebNetBindingConstants.BRIDGE_SUPPORTED_THING_TYPES;
+
+    // ConcurrentHashMap of devices registered to this BridgeHandler
+    // Association is: String ownId -> OpenWebNetThingHandler, with ownId = WHO.WHERE
+    private Map<String, OpenWebNetThingHandler> registeredDevices = new ConcurrentHashMap<>();
+
+    @Nullable
+    protected OpenGateway gateway;
+    private boolean isBusGateway = false;
+
+    private boolean isGatewayConnected = false;
+
+    @Nullable
+    public OpenWebNetDeviceDiscoveryService deviceDiscoveryService;
+    private boolean searchingGatewayDevices = false; // devices search is in progress on gateway
+    private boolean scanIsActive = false; // a device scan has been activated by OpenWebNetDeviceDiscoveryService;
+    private boolean discoveryByActivation = CONFIG_GATEWAY_DEFAULT_DISCVERY_ACTIVATION; // discover BUS devices when
+                                                                                        // they are activated also when
+                                                                                        // a device
+    // scan is not active
+    @Nullable
+    private OpenNewDeviceListener deviceDiscoveryListener = null;
+
+    public OpenWebNetBridgeHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Nullable
+    public OpenGateway getGateway() {
+        return gateway;
+    }
+
+    public boolean isBusGateway() {
+        return isBusGateway;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("==OWN== BridgeHandler.initialize() ");
+
+        ThingTypeUID thingType = getThing().getThingTypeUID();
+        logger.debug("==OWN== Bridge type: {}", thingType);
+
+        if (thingType.equals(THING_TYPE_ZB_GATEWAY)) {
+            initZigBeeGateway();
+        } else {
+            initBusGateway();
+            isBusGateway = true;
+        }
+
+        gateway.subscribe(this);
+        if (gateway.isConnected()) { // gateway is already connected, device can go ONLINE
+            isGatewayConnected = true;
+            logger.info("==OWN== ------------------- ALREADY CONNECTED -> setting status to ONLINE");
```

Comment:
Is logging the prefix `===OWN===` necessary? Logging includes already the class' name.